recipes/expat/all/conanfile.py

```diff
@@ -25,87 +27,69 @@ class ExpatConan(ConanFile):
         "char_type": "char",
     }
 
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if tools.Version(self.version) < "2.2.8":
+        if Version(self.version) < "2.2.8":
             del self.options.char_type
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
+        if not is_msvc(self):
+            del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
```

Comment:
Honestly, conan v2 should lead to less boilerplate, not the opposite.

Reply:
IMO this is something to investigate and could be fixed without problem as conan v2 is on an early stage

Reply:
this is the expected behavior: https://github.com/conan-io/conan/issues/11683 Can be changed to try/catch, but I am leaving it as it is.